library/java/io/envoyproxy/envoymobile/engine/AndroidEngineImpl.java

```diff
@@ -48,6 +48,11 @@ public void flushStats() {
     envoyEngine.flushStats();
   }
 
+  @Override
+  public String dumpStats(int timeout_ms) {
```

Comment:
On iOS it's a convention to use `TimeInterval` for cases like this (it's a double that expresses time in seconds) -> you could call it `timeout` then. 

That being said, I think that we should get rid of this argument altogether. Let's come up with a reasonable value and hardcode it in the engine itself so that it's consistent between iOS and Android. We can then document the method telling its users that it's a blocking one and can take significant amount of time (and hence it's reasonable to call it from a non-main thread).

Otherwise, I think that we will end up in a situation in which the caller needs to decide on a timeout that's not obvious (as in, should this be 5s as for network requests or 100ms because it's local).

Reply:
Maybe before you make a change (if you do not have a strong opinion) I will let @goaway share his opinion here in case he thinks we should keep this parameter